lib/cooked_post_processor.rb

```diff
@@ -81,11 +81,70 @@ def post_process_images
     return if images.blank?
 
     images.each do |img|
+      next if large_images.include?(img["src"]) && add_large_image_placeholder!(img)
+
       limit_size!(img)
       convert_to_link!(img)
     end
   end
 
+  def add_large_image_placeholder!(img)
```

Comment:
Nokogiri API is way too verbose. Is there a way we could inject HTML instead?

Reply:
Yes. But we already using it in same class to create lightbox and I thought it's more clean comparing to raw HTML. I guess you feel like those repeated lines of Nokogiri are not necessary here. So I refactored everything by common methods. Also I changed some existing code using those methods. Please have a look now.